test-case/latency-baseline.sh

```diff
@@ -0,0 +1,228 @@
+
+OPT_NAME['r']='rate'            OPT_DESC['r']='Sample rate to try latency with'
+OPT_HAS_ARG['r']=1              OPT_VAL['r']=48000
+
+OPT_NAME['S']='shorts'          OPT_DESC['S']='Try to use 16-bit samples instead of 32-bit, if possible.'
+OPT_HAS_ARG['S']=0              OPT_VAL['S']=0
+
+OPT_NAME['p']='port_p'          OPT_DESC['p']='Jack playback port with loopback. Example: system:playback_1'
+OPT_HAS_ARG['p']=1              OPT_VAL['p']=''
+
+OPT_NAME['c']='port_c'          OPT_DESC['c']='Jack capture port with loopback. Example: system:capture_1'
+OPT_HAS_ARG['c']=1              OPT_VAL['c']=''
+
+OPT_NAME['s']='sof-logger'      OPT_DESC['s']="Open sof-logger trace the data will store at $LOG_ROOT"
+OPT_HAS_ARG['s']=0              OPT_VAL['s']=1
+
+OPT_NAME['v']='verbose'         OPT_DESC['v']='Verbose logging.'
+OPT_HAS_ARG['v']=0              OPT_VAL['v']=0
+
+func_opt_parse_option "$@"
+
+setup_kernel_check_point
+
+# shellcheck disable=SC2016
+AWK_SCRIPT='
+function min(in_array)
+{
+  min_value = "N/A"
+  if (! isarray(in_array) || length(in_array) == 0) return min_value
+  for(idx in in_array) {
+    if (min_value == "N/A" || in_array[idx] < min_value) {
+        min_value = in_array[idx]
+    }
+  }
+  return min_value
+}
+
+function max(in_array)
+{
+  max_value = "N/A"
+  if (! isarray(in_array) || length(in_array) == 0) return max_value
+  for(idx in in_array) {
+    if (max_value == "N/A" || in_array[idx] > max_value) {
+        max_value = in_array[idx]
+    }
+  }
+  return max_value
+}
+
+function sum(in_array)
+{
+  if (! isarray(in_array) || length(in_array) == 0) return 0
+  sum_items=0
+  for(idx in in_array) {
+    sum_items += in_array[idx]
+  }
+  return sum_items
+}
+
+function stddev(in_array)
+{
+  if (! isarray(in_array) || length(in_array) == 0) return -1
+  sum_items=0
+  cnt_items=0
+  for(idx in in_array) {
+    sum_items += in_array[idx]
+    cnt_items += 1
+  }
+  avg = sum_items / cnt_items
+  dev = 0
+  for(idx in in_array) dev += (in_array[idx] - avg)^2
+  return sqrt(dev/(cnt_items - 1))
+}
+
+/^[ ]*[0-9.]+ frames[ ]+[0-9.]+ ms total roundtrip latency/ {
+  sum_frames+=$1
+  sum_ms+=$3
+  latency_frames[NR]=$1
+  latency_ms[NR]=$3
+}
+END {
+  if (length(latency_frames) !=0 && length(latency_ms) != 0) {
+    printf("\"metrics\":[")
+    printf("{\"name\":\"roundtrip latency\", ")
+    printf("\"probes\":%d, ", length(latency_frames))
+    printf("\"avg_frames\":%0.3f, ", (length(latency_frames) ? sum(latency_frames) / length(latency_frames) : 0))
+    printf("\"min_frames\":%0.3f, \"max_frames\":%0.3f, ", min(latency_frames), max(latency_frames))
+    printf("\"avg_ms\":%0.3f, ", (length(latency_ms) ? sum(latency_ms) / length(latency_ms) : 0))
+    printf("\"min_ms\":%0.3f, \"max_ms\":%0.3f, ", min(latency_ms), max(latency_ms))
+    printf("\"stdev_frames\":%0.6f, \"stdev_ms\":%0.6f }", stddev(latency_frames), stddev(latency_ms))
+    printf("]")
+    fflush()
+  }
+}
+'
+opts_json()
```

Comment:
Should this be in case-lib/lib.sh?


```suggestion
opts2json()
```

Reply:
moved to `case-lib/opt.sh`